Library/ViewModels/DiscoveryProjectCardViewModelTests.swift

```diff
@@ -119,4 +123,120 @@ final class DiscoveryProjectCardViewModelTests: TestCase {
 
     self.projectImageUrlString.assertValues(["http://www.kickstarter.com/full.jpg"])
   }
+
+  func testLoadProjectTags_ProjectIsPWL_CategoryIsRoot() {
+    let project = Project.template
+      |> \.staffPick .~ true
+      |> \.category .~ .illustration
+
+    self.loadProjectTags.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, .art, nil))
+
+    self.loadProjectTags.assertValues([
+      [
+        DiscoveryProjectTagPillCellValue(
+          type: .green,
+          tagIconImageName: "icon--small-k",
+          tagLabelText: "Projects We Love"
+        ),
+        DiscoveryProjectTagPillCellValue(
+          type: .grey,
+          tagIconImageName: "icon--compass",
+          tagLabelText: "Illustration"
+        )
+      ]
+    ])
+  }
+
+  func testLoadProjectTags_CategoryIsNil() {
+    let project = Project.template
+      |> \.staffPick .~ true
+      |> \.category .~ .illustration
+
+    self.loadProjectTags.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, nil, nil))
+
+    self.loadProjectTags.assertValues([
+      [
+        DiscoveryProjectTagPillCellValue(
+          type: .green,
+          tagIconImageName: "icon--small-k",
+          tagLabelText: "Projects We Love"
+        ),
+        DiscoveryProjectTagPillCellValue(
+          type: .grey,
+          tagIconImageName: "icon--compass",
+          tagLabelText: "Illustration"
+        )
+      ]
+    ])
+  }
+
+  func testLoadProjectTags_CategoryMatchesProjectCategory() {
+    let project = Project.template
+      |> \.staffPick .~ true
+      |> \.category .~ .illustration
+
+    self.loadProjectTags.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, .illustration, nil))
+
+    self.loadProjectTags.assertValues([
+      [DiscoveryProjectTagPillCellValue(
+        type: .green,
+        tagIconImageName: "icon--small-k",
+        tagLabelText: "Projects We Love"
+      )]
+    ], "Does not show subcategory tag when filtered category is the same as project subcategory")
+  }
+
+  func testTagsCollectionViewHidden_WhenProjectIsNotStaffPick_CategoryTagShouldShow() {
+    let project = Project.template
+      |> \.staffPick .~ false
+      |> \.category .~ .illustration
+
+    self.tagsCollectionViewHidden.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, .art, nil))
+
+    self.tagsCollectionViewHidden.assertValues([false])
+  }
+
+  func testTagsCollectionViewHidden_WhenProjectIsNotStaffPick_CategoryTagShouldNotShow() {
+    let project = Project.template
+      |> \.staffPick .~ false
+      |> \.category .~ .illustration
+
+    self.tagsCollectionViewHidden.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, .illustration, nil))
+
+    self.tagsCollectionViewHidden.assertValues([true])
+  }
+
+  func testTagsCollectionViewHidden_WhenProjectIsStaffPick_CategoryTagShouldShow() {
```

Comment:
Is the name of this test correct? `TagsCollectionViewHidden` but `CategoryTagShouldShow`?

Reply:
I usually start test names with the name of the output being tested, then follow with the case being tested. Does that make sense?